packages/core/src/tracing/instrumentation/database/db2.js

```diff
@@ -48,12 +48,31 @@ function instrument(db2) {
   shimmer.wrap(db2.Database.prototype, 'prepareSync', instrumentPrepareSync);
 }
 
+/**
+ * We need to set `skipIsTracing` check for db2, because of e.g. prepare
+ * statements or transactions. We don't always know if we are tracing or not.
+ */
 function skipTracing(ignoreClsTracing = false) {
-  // CASES: instrumentation is disabled, db call is disabled via suppress header
-  return (
-    cls.skipExitTracing({ isActive, skipIsTracing: true, skipParentSpanCheck: true }) ||
-    (ignoreClsTracing ? false : !cls.isTracing())
-  );
+  const skipExitResult = cls.skipExitTracing({
+    isActive,
+    skipIsTracing: true,
+    skipParentSpanCheck: true,
+    extendedResponse: true
+  });
+
+  const isTracing = cls.isTracing();
+
+  // CASE: `cls.skipExitTracing` decided to skip, respect that
+  if (skipExitResult.skip) return true;
+
+  // CASE: the target parent function wants us to ignore cls.isTracing
+  //       because e.g. we don't know if we trace (parent got lost)
+  if (ignoreClsTracing) return false;
+
+  // CASE: We ignore if tracing or not for `allowRootExitSpan`. See cls file.
+  if (skipExitResult.allowRootExitSpan) return false;
```

Comment:
@kirrg001 

Maybe we can remove this. Please confirm if there is any other specific case for this

Reply:
> // CASE: We ignore if tracing or not for `allowRootExitSpan`. See cls file.

IMO we need to keep this, because this function `skipTracing` skips checking for `isTracing`.
And for `allowRootExitSpan` we need to ignore this check.